citesphere/src/main/java/edu/asu/diging/citesphere/api/v1/user/JobInfoController.java

```diff
@@ -62,16 +64,19 @@ public ResponseEntity<String> getProfile(@RequestHeader HttpHeaders headers) {
         // FIXME: ugly, needs better solution
         if (job instanceof IUploadJob) {
             node.put("groupId", ((IUploadJob)job).getCitationGroup());
-        }
-        if (job instanceof IExportJob) {
+        } else if (job instanceof IExportJob) {
             IExportTask exportTask = exportTaskManager.get(((IExportJob)job).getTaskId());
             node.put("groupId", exportTask.getGroupId());
             node.put("collectionId", exportTask.getCollectionId());
             node.put("exportType", exportTask.getExportType().name());
             node.put("taskId", exportTask.getId());
+        } else if (job instanceof IImportCrossrefJob) {
+            ArrayNode doisNode = mapper.createArrayNode();
+            ((IImportCrossrefJob)job).getDois().forEach(d -> doisNode.add(d));
+            node.set("dois", doisNode);
+            node.put("groupId", ((IImportCrossrefJob)job).getCitationGroup());
```

Comment:
I think it's time to clean this up. let's create a map that maps object type to methods that are then called here.